apps/video/test/marionette/play_multiple_types_test.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+var assert = require('assert'),
+    Video = require('./lib/video');
+
+marionette('play various types of videos', function() {
+
+  var app, client, selectors;
+
+  client = marionette.client({
+    profile: {
+      prefs: {
+        'device.storage.enabled': true,
+        'device.storage.testing': true,
+        'device.storage.prompt.testing': true
+      }
+    }
+  });
+
+  setup(function() {
+    // Remove all files in temp device storage.
+    client.fileManager.removeAllFiles();
+    selectors = Video.Selector;
+    app = new Video(client);
+  });
+
+  function verifyVideo(type, done) {
+    var title = 'VID_0001';
+    var filename = title + '.' + type;
+
+    client.fileManager.add({
+      type: 'videos',
+      filePath: 'test_media/Movies/' + filename
+    });
+    app.launch();
+    app.tapThumbnail(0);
+    // To avoid timing issues with the controls
+    // being auto-hidden when the video is first
+    // played, wait for controls to be hidden, then
+    // show them, by tapping on the video player.
+    setTimeout(function() {
+      app.showControls();
+      // Verify the video has begun playing and the
+      // title has been read properly.
+      var elapsedTime = app.elapsedTime();
+      assert(elapsedTime > '00:00');
+      var titleFromVideo = app.videoTitle();
+      assert(titleFromVideo, title);
+      done();
+    }, 500);
+  }
+
+  test('play mp4 video', function(done) {
+    verifyVideo('mp4', done);
+  });
+  test('play ogg video', function(done) {
+    verifyVideo('ogg', done);
+  });
+  test('play ogv video', function(done) {
+    verifyVideo('ogv', done);
+  });
+  test('play webm video', function(done) {
+    verifyVideo('webm', done);
+  });
```

Comment:
If you can solve the waiting for thumbnails to appear issue described above, then you could refactor this to run one test with four videos and four thumbnails.  Does marionette do a full reboot between each test?  If you can play four videos in one test that is probably much faster than four separate tests. But I'm only suggesting you do that if it is easy and robust.